packages/dds/tree/src/test/cursorTestSuite.ts

```diff
@@ -489,14 +493,28 @@ function checkTraversal(cursor: ITreeCursor) {
         assert(!fieldLengths.has(key), "no duplicate keys");
         fieldLengths.set(cursor.getFieldKey(), expectedFieldLength);
         assert(expectedFieldLength > 0, "only non empty fields should show up in field iteration");
+        assert(compareFieldUpPaths(cursor.getFieldPath(), { field: key, parent: path }));
+
+        // Check that iterating nodes withing field works as expected.
         let actualChildNodesTraversed = 0;
         for (let inNode = cursor.firstNode(); inNode; inNode = cursor.nextNode()) {
             assert(cursor.chunkStart <= actualChildNodesTraversed);
             assert(cursor.chunkLength > actualChildNodesTraversed - cursor.chunkStart);
             assert(cursor.chunkLength + cursor.chunkStart <= expectedFieldLength);
+
+            // Make sure down+up navigation gets back to where it started.
+            // Testing this explicitly here before recursing makes debugging issues with this easier.
+            assert.equal(cursor.fieldIndex, actualChildNodesTraversed);
+            cursor.enterField(EmptyKey);
+            cursor.exitField();
```

Comment:
What does this test? IIRC, doesn't this just push EmptyKey to the navigation stack and then pop it without doing any sort of validation?